nbgrader/apps/assignapp.py

```diff
@@ -12,103 +13,152 @@
+        read in according to:
 
-        > nbgrader assign "Problem 1.ipynb" --save-cells --assignment="Problem Set 1"
+            source/./{assignment_id}/{notebook_id}.ipynb
 
-        You can additionally specifiy the database name with --db:
+        and saved according to:
 
-        > nbgrader assign "Problem 1.ipynb" --save-cells --assignment="Problem Set 1" --db=myclass
+            release/./{assignment_id}/{notebook_id}.ipynb
 
         """
     ))
 
-    save_cells = Bool(False, config=True, help="Save information about grade cells into the database.")
-
-    def _classes_default(self):
-        classes = super(AssignApp, self)._classes_default()
-        classes.extend([
-            IncludeHeaderFooter,
-            LockCells,
-            ClearSolutions,
-            ClearOutputPreprocessor,
-            CheckCellMetadata,
-            ComputeChecksums,
-            SaveCells
-        ])
-        return classes
+    create_assignment = Bool(
+        False, config=True,
+        help=dedent(
+            """
+            Whether to create the assignment at runtime if it does not
+            already exist.
+            """
+        )
+    )
+
+    nbgrader_input_step_name = Unicode(
+        "source",
+        config=True,
+        help=dedent(
+            """
+            The input directory for this step of the grading process. This
+            corresponds to the `nbgrader_step` variable in the path defined by
+            `NbGraderConfig.directory_structure`.
+            """
+        )
+    )
+    nbgrader_output_step_name = Unicode(
+        "release",
+        config=True,
+        help=dedent(
+            """
+            The output directory for this step of the grading process. This
+            corresponds to the `nbgrader_step` variable in the path defined by
+            `NbGraderConfig.directory_structure`.
+            """
+        )
+    )
+
+    export_format = 'notebook'
+
+    preprocessors = List([
+        IncludeHeaderFooter,
+        LockCells,
+        ClearSolutions,
+        ClearOutput,
+        CheckCellMetadata,
+        ComputeChecksums,
+        SaveCells
+    ])
 
     def build_extra_config(self):
-        self.extra_config = Config()
-        self.extra_config.Exporter.preprocessors = [
-            'nbgrader.preprocessors.IncludeHeaderFooter',
-            'nbgrader.preprocessors.LockCells',
-            'nbgrader.preprocessors.ClearSolutions',
-            'IPython.nbconvert.preprocessors.ClearOutputPreprocessor',
-            'nbgrader.preprocessors.CheckCellMetadata',
-            'nbgrader.preprocessors.ComputeChecksums'
-        ]
-        if self.save_cells:
-            self.extra_config.Exporter.preprocessors.append(
-                'nbgrader.preprocessors.SaveCells'
-            )
-        self.config.merge(self.extra_config)
+        extra_config = super(AssignApp, self).build_extra_config()
+        extra_config.NbGraderConfig.student_id = '.'
```

Comment:
It's not a regex -- it's the actual value that gets put into the format string. It makes it so that directory isn't included, essentially. e.g. if the directory structure is `{nbgrader_step}/{student_id}/{assignment_id}/`, then for the assign step it will actually look for stuff in `source/./{assignment_id}/` and save stuff to `release/./{assignment_id}/`